### DISCUSSION TITLE
mysql-client doesn't work after upgrade

### DISCUSSION THREAD

**Discussion by midgard** [Homebrew/discussions#5547] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1e560120a65be3b3c7b8a42083d54f795e1d0ec4
Last commit: 2 days ago
Core tap JSON: 07 Aug 14:33 UTC
Core cask tap JSON: 07 Aug 14:33 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libakisp11.dylib
```

### Description of issue

mysql-client upgraded, and my python script doesn't work now.

`ImportError: dlopen(/Users/kronokrator/PycharmProjects/apt2-be/venv/lib/python3.12/site-packages/MySQLdb/_mysql.cpython-312-darwin.so, 0x0002): Library not loaded: /opt/homebrew/opt/mysql-client/lib/libmysqlclient.23.dylib
  Referenced from: <8ADCE65B-9620-3A41-97D7-777E35921CE5> /Users/kronokrator/PycharmProjects/apt2-be/venv/lib/python3.12/site-packages/MySQLdb/_mysql.cpython-312-darwin.so
  Reason: tried: '/opt/homebrew/opt/mysql-client/lib/libmysqlclient.23.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/mysql-client/lib/libmysqlclient.23.dylib' (no such file), '/opt/homebrew/opt/mysql-client/lib/libmysqlclient.23.dylib' (no such file), '/opt/homebrew/Cellar/mysql-client/9.0.1/lib/libmysqlclient.23.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/mysql-client/9.0.1/lib/libmysqlclient.23.dylib' (no such file), '/opt/homebrew/Cellar/mysql-client/9.0.1/lib/libmysqlclient.23.dylib' (no such file)
`

**Comment by SMillerDev** [comment]
Yeah, it seems your script or one of its dependencies hardcoded the path to the MySQL client library. That path will need to be updated now that the library is updated.

**Comment by ritbl** [comment, marked as answer]
I had the same issue, I needed to reinstall mysqlclient with no-cache option.

```
export PKG_CONFIG_PATH="$(brew --prefix)/opt/mysql-client/lib/pkgconfig"
pip install --force-reinstall --no-cache mysqlclient
```

**Reply by JwabeNero** [reply]
Man, after an entire day, like the entire day, trying to solve this F&$K UP, which I didn't understand where it came from, these two simple lines saved me!!! Thank you!!!!

**Reply by ninadvopa** [reply]
Thank you so much!! This saved my day! I faced similar issue mentioned in the description and these two lines solved it. ( force installation with no cache)

**Reply by cbeav** [reply]
This dude rocks

**Reply by Mereep** [reply]
this works, unless you upgraded from mysql-client 8 to 9, which will then very likely fail due to missing mysql-password, see https://github.com/Homebrew/homebrew-core/issues/180498
